src/components/templates/Playa/AvatarLayer.tsx

```diff
@@ -125,43 +127,69 @@ const AvatarLayer: React.FunctionComponent<PropsType> = ({
     () => (
       <MyAvatar
         serverSentState={myServerSentState}
-        zoom={zoom}
         walkMode={walkMode}
         sendUpdatedState={sendUpdatedState}
         setMyLocation={setMyLocation}
       />
     ),
-    [myServerSentState, zoom, walkMode, sendUpdatedState, setMyLocation]
+    [myServerSentState, walkMode, sendUpdatedState, setMyLocation]
   );
 
-  return useMemo(
-    () => (
-      <>
-        {myAvatar}
-        {Object.keys(userStateMap)
-          .sort()
-          .filter(
-            (uid) =>
-              user?.uid !== uid &&
-              !!partygoers.find((partygoer) => partygoer.id === uid)
-          )
-          .map((uid) => (
+  const avatars = useMemo(
+    () =>
+      Object.keys(userStateMap)
+        .sort()
+        .filter(
+          (uid) =>
+            user?.uid !== uid &&
+            !!partygoers.find((partygoer) => partygoer.id === uid)
+        )
+        .map((uid) => {
+          const avatarUser = partygoers.find(
+            (partygoer) => partygoer.id === uid
+          );
+          return (
             <Avatar
-              user={partygoers.find((partygoer) => partygoer.id === uid)}
+              user={avatarUser}
               state={userStateMap[uid]}
-              zoom={zoom}
               setSelectedUserProfile={setSelectedUserProfile}
+              setHoveredUser={setHoveredUser}
+              setHovered={setHovered}
+              hoveredRef={hoveredRef}
               key={uid}
             />
-          ))}
+          );
+        }),
+    [partygoers, user, userStateMap]
+  );
+
+  return useMemo(
+    () => (
+      <>
+        {myAvatar}
+        {avatars}
+        <Overlay target={hoveredRef.current} show={hovered}>
```

Comment:
could just use hoveredUser and show={!!hoveredUser}. Will need to `setHoveredUser(null)` on mousOut

Reply:
I did that, but unsetting the hovered user makes the fading-out overlay lose its user and such, it looks super janky. So I re-added the boolean